dfsclient/src/main/java/org/hpccsystems/dfs/client/CompiledFieldFilter.java

```diff
@@ -0,0 +1,294 @@
+        }
+        else
+        {
+            // For more than two object we either have a range between
+            // 2 values or a set. In the case of the the range they should
+            // already be sorted but it cannot hurt to do it again.
+            // In the case of the set we want to find the largest and smallest values
+            // So sort the converted values to find the min and max for our range.
+            Object[] convertedValues = new Object[filterValues.length];
+            for (int i = 0; i < filterValues.length; i++)
+            {
+                convertedValues[i] = convertValue(filterValues[i], fdType);
+            }
+
+            Arrays.sort(convertedValues, (a, b) -> this.compare(a,b));
+
+            lowerValue = convertedValues[0];
+            upperValue = convertedValues[convertedValues.length-1];
+        }
+
+        if (filter.getBound() == FieldFilterRange.Bound.UPPER || filter.getBound() == FieldFilterRange.Bound.NONE)
+        {
+            this.lowerValue = getMinValueForType(fdType);
+        }
+
+        if (filter.getBound() == FieldFilterRange.Bound.LOWER || filter.getBound() == FieldFilterRange.Bound.NONE)
+        {
+            this.upperValue = getMaxValueForType(fdType);
+        }
+
+        if (filter.getLeftOpen())
+        {
+            this.lowerComparisonType = 1;
+        }
+
+        if (filter.getRightOpen())
+        {
+            this.upperComparisonType = 1;
+        }
+    }
+
+    public int getFieldIndex()
+    {
+        return fieldIndex;
+    }
+
+    private int compare(Object a, Object b) 
+    {
+        switch (fieldType)
+        {
+            case INTEGER:
+                {
+                    Long la = (Long) a;
+                    Long lb = (Long) b;
+                    return la.compareTo(lb);
+                }
+            case BOOLEAN:
+                {
+                    Boolean ba = (Boolean) a;
+                    Boolean bb = (Boolean) b;
+                    return ba.compareTo(bb);
+                }
+            case REAL:
+                {
+                    Double da = (Double) a;
+                    Double db = (Double) b;
+                    return da.compareTo(db);
+                }
+            case DECIMAL:
+                {
+                    BigDecimal da = (BigDecimal) a;
+                    BigDecimal db = (BigDecimal) b;
+                    return da.compareTo(db);
+                }
+            case STRING:
+                {
+                    String sa = (String) a;
+                    String sb = (String) b;
+
+                    if (comparisonLength < 0)
+                    {
+                        return sa.compareTo(sb);
+                    }
+                    else
+                    {
+                        return sa.substring(0,comparisonLength).compareTo(sb.substring(0,comparisonLength));
+                    }
+                }
+            default:
+                return 0;
```

Comment:
this return value seems misleading, if we can't compare this type, perhaps a exception is appropriate